narwhals/_pandas_like/series.py

```diff
@@ -762,6 +762,13 @@ def mode(self: Self) -> Self:
     def __iter__(self: Self) -> Iterator[Any]:
         yield from self._native_series.__iter__()
 
+    def is_finite(self: Self) -> Self:
+        import numpy as np  # ignore-banned-import
+
+        return self._from_native_series(
+            np.isfinite(self._native_series) & ~self._native_series.isna()
+        )
```

Comment:
Here is a opinionated choice that na is not finite

Reply:
🤔 no sure, wouldn't we want to preserve null values?

Reply:
Behavior is different for different pandas backend dtype. Let me come back with an example

Reply:
hmmm actually, for classical pandas types, we wouldn't have the option of returning a nullable boolean (if we want to preserve the dtype backend)

🤔 gonna think about this a little longer

Reply:
These would be the output:

```py
data = [float("nan"), float("inf"), 2.0, None]

s = pd.Series(data)
np.isfinite(s)

0    False
1    False
2     True
3    False
dtype: bool
```

```py
np.isfinite(s.convert_dtypes(dtype_backend="numpy_nullable"))

0     <NA>
1    False
2     True
3     <NA>
dtype: boolean
```

```py
np.isfinite(s.convert_dtypes(dtype_backend="pyarrow"))

0    False
1    False
2     True
3    False
dtype: bool
```

Reply:
@MarcoGorelli considering the new page we have on booleans, how would you move forward with this?
I am asking just to discuss it, I am ok with keeping the inconsistencies between different pandas dtype backends, and let the use handle those. At the same time, I would aim at some sort of unification towards polars behavior, although in this specific context it seems unfeasible

Reply:
Looks good, I will resolve conflicts with main and adjust